src/users/views.py

```diff
@@ -135,10 +137,48 @@ def get_object(self):
 
 
 class MentorsSearchView(generics.ListAPIView):
-    queryset = Mentor.objects.all()
+    """
+    View for finding a mentor by major
+    """
+    queryset = Mentor.objects.all().filter(active=True)
     serializer_class = MentorSerializer
 
     def filter_queryset(self, queryset):
         major = self.request.query_params['major']
         return queryset.filter(major__name=major)
 
+
+class MentorView(APIView):
+    """
+    Currently Unneeded: View Mentor By ID
+    """
+    serializer_class = MentorSerializer
+    def get(self, request, *args, **kwargs):
+        mentor_id = int(self.kwargs['mentor_id'])
+        mentor = get_object_or_404(Mentor, id=mentor_id)
+        return Response(MentorSerializer(mentor).data)
+
+
+class OwnMentorView(generics.RetrieveUpdateDestroyAPIView):
+    """
+    View for turning mentor status on and off
+    """
+    serializer_class = MentorSerializer
+    def get_object(self):
+        return get_object_or_404(Mentor, profile__user=self.request.user)
+    serializer_class = MentorSerializer
+    def post (self,request):
+
+        profile_id = self.request.user.profile.id
+        profile = get_object_or_404(Profile, id=profile_id)
+        try:
+            new_mentor = get_object_or_404(Mentor, profile__user=self.request.user)
+            new_mentor.active = True
+        except:
```

Comment:
never use empty excepts.  I would write it as follows:

```python
mentor_request = Mentor.objects.filter(profile__user=self.request.user)
if not mentor_request.exists():
    mentor = Mentor(profile=profile, active=True)
else:
    mentor = mentor_request[0]
mentor.save()

return Response(MentorSerializer(new_mentor).data)
```

This still only uses one request to get the mentor object, and no excepts.

Except is bad because if you have any unexpected exceptions, they would be eaten and impossible to debug.  

Let's say that `get_object_or_404(Mentor, profile__user=self.request.user)` throws a database connection error.  we wouldn't be able to see that in the debug output at all